Fia/baekjoon/Fia11725.py

```diff
@@ -0,0 +1,39 @@
+import sys
+import collections
+
+N = int(sys.stdin.readline())
+trees = collections.defaultdict(list)
+
+for _ in range(N - 1):
+    node1, node2 = map(int, sys.stdin.readline().split())
+    trees[node1].append(node2)
+    trees[node2].append(node1)
+
+parents = [0] * (N + 1)
+
+# dfs 풀이   440ms
+# visited = set()
+#
+#
+# def traversal(root):
+#     visited.add(root)
+#     for node in trees[root]:
+#         if parents[node] == 0:
+#             parents[node] = root
+#     for node in trees[root]:
```

Comment:
왜 for 문이 2개가 필요한가요? (질문)

Reply:
잉 그렇네요 없어도 될 것 같아요 (대답)
```suggestion
def traversal(root):
    visited.add(root)
    for node in trees[root]:
        if parents[node] == 0:
            parents[node] = root
        if node not in visited:
            traversal(node)
```

Reply:
ㅋㅋㅋㅋㅋㅋㅋ (질문) (대답)